src/pip/_internal/index/package_finder.py

```diff
@@ -930,6 +931,14 @@ def find_all_candidates(self, project_name: str) -> List[InstallationCandidate]:
         )
         page_candidates = list(page_candidates_it)
 
+        # Since candidates_from_page does not get used to process file sources
+        # from find-links, manually inject evaluate_links calls here.
+        _links = [
+            Link(path_to_url(fl.variants_json))
+            for fl in collected_sources.find_links
+            if fl.variants_json is not None
+        ]
+        self.evaluate_links(link_evaluator, _links)
```

Comment:
Not a very clean way of doing things, but I figured better than nothing given that we're still prototyping. I'd be happy with a more structured solution at some point but I don't think it's urgent.